### ISSUE TITLE
Programatic usage and side-effects

### ISSUE THREAD

**Issue by SkyaTura** [RobinHerbots/jquery.inputmask#2645]
It's awesome how this project is complete in terms of many different things involved in masks, which are well known nightmares for those who has tried to do it manually before.

However, the way HTMLElements are currently being manipulated is somewhat problematic. Although I do understand the context how (and when) this was made, the scenario we have in Web Development nowadays is completely different, and strategies has changed substantially at the point using this lib can directly conflict with some frameworks.

Today, the lib depends on directly mutating DOM elements, which may be dynamically changed many times.

**TL;DR**
Well, I think I don't need to explain it further, but my question/proposal is if it's viable to at least think in a different structure of the project that allow other libraries to use this as a building block to framework specific tools.
